plugins/workflow/category_transition/src/Extension/CategoryTransition.php

```diff
@@ -0,0 +1,239 @@
+    protected function disableCategoryField(Form $form)
+    {
+        // Get the current category ID value
+        $catid = $form->getValue('catid');
+
+        $form->setFieldAttribute('catid', 'readonly', 'true');
+        $form->setFieldAttribute('catid', 'value', $catid);
+    }
+
+
+    /**
+     * Method to handle the workflow transition event.
+     *
+     * @param   WorkflowTransitionEvent  $event  The event object
+     *
+     * @return  void
+     *
+     * @since   DEPLOY_VERSION
+     */
+    public static function onWorkflowAfterTransition(WorkflowTransitionEvent $event): void
+    {
+        $app = Factory::getApplication();
+        $pks = $event->getArgument('pks');
+        $transition = $event->getArgument('transition');
+
+        if (!self::validateTransition($app, $transition)) {
+            return;
+        }
+
+        $options = $transition->options;
+        $categoryId = (int) $options->get('category_id');
+
+        if (empty($pks) || !is_array($pks)) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_TRANSITION_NO_PRIMARY_KEY'), 'error');
+            return;
+        }
+
+        $errors = 0;
+
+        foreach ($pks as $pk) {
+            if (!self::processArticle($app, $pk, $categoryId)) {
+                $errors++;
+            }
+        }
+
+        if ($errors > 0) {
+            $app->enqueueMessage(Text::sprintf('PLG_WORKFLOW_CATEGORY_TRANSITION_ERROR', $errors), 'warning');
+        }
+    }
+
+    /**
+     * Validate the transition object.
+     *
+     * @param   CMSApplicationInterface  $app        The application object
+     * @param   object               $transition The transition object
+     *
+     * @return  bool
+     *
+     * @since   DEPLOY_VERSION
+     */
+    private static function validateTransition($app, $transition): bool
+    {
+        if (!is_object($transition)) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_TRANSITION_INVALID_OBJECT_TYPE'), 'error');
+            return false;
+        }
+
+        if (!($transition->options instanceof Registry)) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_TRANSITION_INVALID_REGISTRY'), 'error');
+            return false;
+        }
+
+        if ($transition->options === null) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_TRANSITION_NO_OPTIONS'), 'error');
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Process the article and update its category.
+     *
+     * @param   CMSApplicationInterface  $app        The application object
+     * @param   int                  $pk         The primary key
+     * @param   int                  $categoryId The category ID
+     *
+     * @return  bool
+     *
+     * @since   DEPLOY_VERSION
+     */
+    private static function processArticle($app, $pk, $categoryId): bool
+    {
+        $result = false;
+
+        try {
+            $articleTable = Table::getInstance('Content');
```

Comment:
Please boot the component and create the article table there

```suggestion
            $articleTable = $this->getApplication()->bootComponent('com_content')->getMVCFactory()->createModel('Content', 'Administrator', ['ignore_request' => true)->getTable('Content');;
```